### PR TITLE
Merge 21.5.1 hotfix into `release/21.6`

### PR DESCRIPTION
**Pull request by mokagio**
What it says on the title. 

As a confirmation of the successful hotfix merge, notice diff equals to that from #19961 minus the `RELEASE-NOTES.txt` amendment and that the `Version.*.xcconfig` files have not changed.

## Next steps

I'll ship a new beta for 21.6. Apart from the fact that we can't  ship betas from non release branches, I like how the merge commit from this PR will track the hotfix integration in the Git history neatly.

### CODE REVIEW

**Review by twstokes — APPROVED** [review]
FWIW it LGTM @mokagio but I understand if we also need to get approval from the Apps Infra team. 👍

**Comment by mokagio** [comment]
> but I understand if we also need to get approval from the Apps Infra team. 👍

Auto-merge already kicked in, but anyways I'd say that for a straightforward PR like this one, anyone's approval is fine. 

Thank you for always helping @twstokes !

**Pull request merged**
